omnigibson/transition_rules.py

```diff
@@ -411,6 +412,49 @@ def modifies_filter_names(self):
         return {self._filter_1_name}
 
 
+class TouchingVolumeCheckCondition(TouchingAnyCondition):
+    def __init__(self, filter_1_name, filter_2_name):
+        '''
+        Checks if filter_2_bodies have a contact point within the collision volume generated from each obj in filter_1
+        '''
+        # super().__init__(filter_2_name, filter_1_name)
+        super().__init__(filter_1_name, filter_2_name)
+
+    def __call__(self, object_candidates):
+        # Keep any object that has non-zero impulses between itself and any of the @filter_2_name's objects
+        objs = []
+
+        # Manually check contact
+        filter_2_bodies = set.union(*(self._filter_2_bodies[obj] for obj in object_candidates[self._filter_2_name]))
+        for obj in object_candidates[self._filter_1_name]:
+            slicer_volume = None
+            for name, link in self.obj.links.items():
+                if 'slicer' in name:
+                    slicer_volume = link
+                    break
+            
+            if slicer_volume is None:
+                continue
+
+            check_in_volume, _ = generate_points_in_volume_checker_function(
+                obj=obj,
+                volume_link=slicer_volume  
+            )
```

Comment:
Ask @cremebrule if repeatedly creating this function incurs a big performance penalty. We could do something clever here where the volume is known and the checker function pointer is held by the SlicerActive state and here we can just query it.

At the very least, add before this line:
```
# TODO: Consider moving this to the SlicerActive state for performance.
```